server/src/LspServer.ts

```diff
@@ -187,21 +188,32 @@ export class LspServer {
   }
 
   async prepareDestination(profileResult: Result<DbtProfileSuccess, DbtProfileError>): Promise<void> {
-    if (profileResult.isOk()) {
-      const bigQueryContextInfo = await BigQueryContext.createContext(profileResult.value);
+    if (
+      profileResult.isOk() &&
+      profileResult.value.dbtProfile &&
+      profileResult.value.type &&
+      profileResult.value.type.valueOf() === DbtProfileType.BigQuery
+    ) {
+      const bigQueryContextInfo = await BigQueryContext.createContext(profileResult.value.dbtProfile, profileResult.value.targetConfig);
       if (bigQueryContextInfo.isOk()) {
         this.bigQueryContext = bigQueryContextInfo.value;
       } else {
-        this.showPrepareDestinationWarning(bigQueryContextInfo.error);
+        this.showCreateContextWarning(bigQueryContextInfo.error);
       }
-    } else {
+    } else if (profileResult.isErr()) {
       this.showPrepareDestinationWarning(profileResult.error.message);
     }
     this.contextInitializedDeferred.resolve();
   }
 
+  showCreateContextWarning(error: string): void {
+    const message = `Unable to initialize BigQuery. ${error}`;
```

Comment:
What will be the exact error shown (what error contains here), may be attach screenshot? And could you clarify what can be the reason of this error?

Reply:
It could be some problems related to BigQuery client initialization. 
<img width="468" alt="image" src="https://user-images.githubusercontent.com/93979055/174819357-81efc8cb-b489-451f-8c95-4c73b4b64355.png">